src/main/scala/com/exasol/cloudetl/kafka/KafkaTopicDataImporter.scala

```diff
@@ -43,17 +42,19 @@ object KafkaTopicDataImporter extends LazyLogging {
     val topic = kafkaProperties.getTopic()
     val topicPartition = new TopicPartition(topic, partitionId)
 
-    val recordDeserialization = kafkaProperties.getRecordFormat() match {
-      case "avro" => AvroDeserialization
-      case "json" => JsonDeserialization
-    }
-    val recordFields = kafkaProperties.getRecordFields()
-    val derserializer = if (kafkaProperties.getSingleColJson()) {
-      recordDeserialization.getSingleColumnJsonDeserializer(kafkaProperties, recordFields)
-    } else {
-      recordDeserialization.getColumnDeserializer(kafkaProperties, recordFields)
-    }
-    val kafkaConsumer = KafkaConsumerFactory(kafkaProperties, derserializer, metadata)
+    val fieldSpecs = FieldParser.get(kafkaProperties.getRecordFields())
+
+    val recordDerse = DeserializationFactory
+      .getSerializers(fieldSpecs, kafkaProperties)
+
+    val kafkaConsumer =
+      KafkaConsumerFactory(
+        kafkaProperties,
+        recordDerse.keyDeserializer,
+        recordDerse.valueDeserializer,
```

Comment:
..as a follow-on to the above suggestion:
```suggestion
        recordDeserializers.keyDeserializer,
        recordDeserializers.valueDeserializer,
```